Open Judge System/OJS.Common/Extensions/FileHelpers.cs

```diff
@@ -56,7 +58,10 @@ public static string FindFirstFileMatchingPattern(string workingDirectory, strin
                     nameof(pattern));
             }
 
-            return ProcessModulePath(files[0]);
+            string discoveredFile = files
+                .OrderByDescending(f => new FileInfo(f).Length).First();
```

Comment:
Add a validation ensuring that there is at least one file found or alternatively return null. Use FirstOrDefault instead of First and ensure that ProcessModulePath is called only if a matching file is found.